design/design.md

```diff
@@ -0,0 +1,175 @@
+# Asteria dApp Design
+
+## Introduction
+This document describes the technical design of the Asteria dApp - the script UTxOs involved, the operations that take place both during the game and in the setup phase, and the necessary validators and minting policies.
+
+There will be a single script UTxO for the `Pot`, several `PelletState` UTxOs and a `ShipState` UTxO for every user. The `Pot` UTxO locks the ada amount paid by each user when creating a ship, and it's position on the board is always assumed to be (0,0). Both `PelletState` and `ShipState` UTxOs have their positions specified in the datum. In order to identify valid game UTxOs, the admin will deposit a special token in the `PelletState` and `Pot` UTxOs when creating them.
+
+Each ship will be identified by a `ShipToken`, with a fixed policy id but a token name of their own. This is the token that is minted by the _Ship minting policy_, described in the validators section.
+
+
+## UTxOs specification
+
+### ShipState UTxO:
+
+>#### Address
+>* Parameterized on admin token, `Pot` validator address and `PelletState` validator address.
+>
+>#### Datum
+>* pos_x: **Int**
+>* pos_y: **Int**
+>* fuel: **Int**
+>* pilot_token: **ByteArray**
+>
+>#### Value
+>* minAda
+>* `ShipToken`
+
+### PelletState UTxO:
+
+>#### Address
+>* Parameterized on the admin token.
+>
+>#### Datum
+>* pos_x: **Int**
+>* pos_y: **Int**
+>* fuel: **Int**
+>* ship_token_policy: **PolicyId**
+>
+>#### Value
+>* minAda
+>* admin token
+
+### Pot UTxO:
+
+>#### Address
+>* Parameterized on the admin token.
+>
+>#### Datum
+>* ship_token_policy: **PolicyId**
+>
+>#### Value
+>* minAda
+>* reward amount.
+>* admin token
+>
+>Note: the reward amount is updated during the course of the game, whenever a new user joins or reaches the pot.
+
+
+## Transactions
+
+### Create Pot UTxO:
+This transaction creates the unique `Pot` UTxO locking min ada and an admin token.
```

Comment:
We should indicate that the tx stores in the datum the ship token policy id. Something like:

This transaction creates...... and an admin token. It stores in the datum the ShipToken policy id for being able to reference it in the validator.